apiserver/client/status.go

```diff
@@ -1062,10 +1082,24 @@ func populateStatusFromStatusInfoAndErr(agent *params.DetailedStatus, statusInfo
 	agent.Since = statusInfo.Since
 }
 
+// contextMachine overloads the Status call to use the cached status values,
+// and delegates everything else to the Machine.
+// TODO: cache presence as well.
```

Comment:
Presence is already in-memory structures, I don't think we need yet-another cache for it.
I do have code that would allow you to ask about the presence information of multiple agents in one channel send and receive, instead of each agent being a separate request.
I just didn't bother updating it, because all the accesses were thisunit.Status() so we didn't have a way to call the bulk presence function.